doc: matter: Fixes for the Matter documentation (#14380)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * Fixed an issue that prevented nRF Toolbox for iOS in version 5.0.9 from controlling the sample using :ref:`nus_service_readme`.
  * Changed the design of the :ref:`matter_lock_sample_wifi_thread_switching` feature so that support for both Matter over Thread and Matter over Wi-Fi is included in a single firmware image.

>   * Added support for Wi-Fi firmware upgrade on external memory, only for the combination of the nRF5340 DK with the nRF7002 EK.

Reviewer: ```suggestion
  * Added support for Wi-Fi firmware upgrade on external memory.
    This is only available for the nRF5340 DK with the nRF7002 EK.
```
